misc/peer-store/src/store.rs

```diff
@@ -0,0 +1,99 @@
+use std::{
+    collections::{HashMap, HashSet},
+    time::SystemTime,
+};
+
+use libp2p_core::{Multiaddr, PeerId};
+
+/// A store that
+/// - keep track of currently connected peers;
+/// - contains all observed addresses of peers;
+pub trait Store {
+    /// Called when a peer connects.
+    fn on_peer_connect(&mut self, peer: &PeerId);

```

Comment:
If we are tracking peer connections too, should we also keep tabs on the `ConnectionId`?

Reply:
We can, but how to store it? I think there can be multiple connections from a single peer, no?